contracts/AutomationBotAggregator.sol

```diff
@@ -76,55 +61,36 @@ contract AutomationBotAggregator {
         return (manager.cdpCan(cdpOwner, cdpId, operator) == 1) || (operator == cdpOwner);
     }
 
-    function getBotAndAggregator()
-        internal
-        view
-        returns (AutomationBot bot, AutomationBotAggregator aggregator)
-    {
-        bot = AutomationBot(serviceRegistry.getRegisteredService(AUTOMATION_BOT_KEY));
-        aggregator = AutomationBotAggregator(
-            serviceRegistry.getRegisteredService(AUTOMATION_BOT_AGGREGATOR_KEY)
-        );
+    function getValidatorAddress(uint16 groupType) public view returns (IValidator) {
+        bytes32 validatorHash = keccak256(abi.encode("Validator", groupType));
+        return IValidator(serviceRegistry.getServiceAddress(validatorHash));
     }
 
     function addTriggerGroup(
         uint16 groupType,
         uint256[] memory replacedTriggerId,
         bytes[] memory triggersData
     ) external onlyDelegate {
-        (AutomationBot bot, AutomationBotAggregator aggregator) = getBotAndAggregator();
-        IValidator validator = IValidator(getValidatorAddress(groupType));
-        require(validator.validate(replacedTriggerId, triggersData), "aggregator/validation-error");
-        (uint256[] memory cdpIds, uint256[] memory triggerTypes) = validator.decode(triggersData);
-        uint256 firstTriggerId = bot.triggersCounter() + 1;
-        uint256[] memory triggerIds = new uint256[](triggersData.length);
-        for (uint256 i = 0; i < triggerTypes.length; i++) {
-            (bool status, ) = address(bot).delegatecall(
-                abi.encodeWithSelector(
-                    bot.addTrigger.selector,
-                    cdpIds[i],
-                    triggerTypes[i],
-                    replacedTriggerId[i],
-                    triggersData[i]
-                )
-            );
-            triggerIds[i] = firstTriggerId + i;
-            require(status, "aggregator/add-trigger-failed");
-        }
+        ManagerLike manager = ManagerLike(serviceRegistry.getRegisteredService(CDP_MANAGER_KEY));
 
-        aggregator.addRecord(cdpIds[0], groupType, triggerIds);
+        (uint256[] memory cdpIds, ) = getValidatorAddress(groupType).decode(triggersData);
+        uint256 cdpId = cdpIds[0];
+        address aggregator = serviceRegistry.getRegisteredService(AUTOMATION_BOT_AGGREGATOR_KEY);
+        if (!isCdpAllowed(cdpId, aggregator, manager)) {
+            manager.cdpAllow(cdpId, aggregator, 1);
```

Comment:
same operation needs to be done to automationBot if it lacks cdpAllow otherwise trigger will fail on execution

Reply:
fixed